src/client_side.cc

```diff
@@ -3182,7 +3182,9 @@ ConnStateData::initiateTunneledRequest(HttpRequest::Pointer const &cause, const
         static char ip[MAX_IPSTRLEN];
         connectHost = clientConnection->local.toStr(ip, sizeof(ip));
         connectPort = clientConnection->local.port();
-    } else {
+    }
+
+    if (!connectPort) {
```

Comment:
This minor change prevents possible bugs (in the official code) where we may try to tunnel to a zero port.